### PR TITLE
Use .data pronoun to prevent global variable warning in R CMD check

### PR DESCRIPTION
**Pull request by JosiahParry**
This PR resolves a note about a global variable by using `.data` prefix as well as modifies the vendoring test to no longer print crate versions but rather just the crate names themselves.

### CODE REVIEW

File: tests/testthat/test-use_cran_defaults.R [hunk 1]
```diff
@@ -35,6 +35,6 @@ test_that("vendor_pkgs() vendors dependencies", {
 
   package_versions <- vendor_pkgs(path, quiet = TRUE)
   expect_snapshot(cat_file("src", "rust", "vendor-config.toml"))
-  expect_snapshot(package_versions)
```
Comment (Ilia-Kosenkov) [inline tests/testthat/test-use_cran_defaults.R:38]: My last PR actually fixed versions here, can you roll this change back (and also snapshot)?

File: R/cran-compliance.R [hunk 2]
```diff
@@ -191,7 +191,7 @@ vendor_pkgs <- function(path = ".", quiet = FALSE, overwrite = NULL) {
     rlang::set_names(c("source", "crate", "version")) %>%
     dplyr::filter(!is.na(source)) %>%
     dplyr::select(-source) %>%
-    dplyr::arrange(crate) # nolint: object_usage_linter
+    dplyr::arrange(.data$crate) # nolint: object_usage_linter
```
Comment (Ilia-Kosenkov) [inline R/cran-compliance.R:194]: I think `nolint` is no longer needed because of the `.data` pronoun.

**Comment by JosiahParry** [comment]
oooof course we have a conflict. I think i want to keep versions from this branch.

**Comment by Ilia-Kosenkov** [comment]
The PR titles seems to not reflect the changes, can you align it?

**Comment by JosiahParry** [comment]
I've changed the title. Please let me know if it works.

**Review by Ilia-Kosenkov — APPROVED** [review]

**Pull request merged**